### DISCUSSION TITLE
About Improving "Colour" Documentation.

### DISCUSSION THREAD

**Discussion by thomasmansencal** [colour-science/colour#682] [Ideas]
This article has some good information that could be useful to apply to our documentation: https://www.divio.com/blog/documentation/. The gist of it is a clear delineation between those 4 pillars:

## TUTORIALS
A tutorial:

- is **learning-oriented**
- allows the newcomer to get started
- is a lesson

_Analogy: teaching a small child how to cook_

## HOW-TO GUIDES
A how-to guide:

- is **goal-oriented**
- shows how to solve a specific problem
- is a series of steps

_Analogy: a recipe in a cookery book_

## EXPLANATION
An explanation:

- is **understanding-oriented**
- explains
- provides background and context

_Analogy: an article on culinary social history_

## REFERENCE
A reference guide:
 
- is **information-oriented**
- describes the machinery
- is accurate and complete

_Analogy: a reference encyclopaedia article_

**Comment by thomasmansencal** [comment]
I re-organised the [README](https://github.com/colour-science/colour/blob/develop/README.rst) and the [Manual](https://colour.readthedocs.io/en/develop/) to delineate better the various sections as per the above. It was also the occasion to create a [How-To Guide](https://colab.research.google.com/drive/1NRcdXSCshivkwoU2nieCvC3y14fx1X4X#offline=true&sandboxMode=true).

More to come!

**Comment by Myndex** [comment]
H @KelSolaar  Hi Thomas, and thank you,

*"The Grand Unified Theory of Documentation"*

I struggle with this a lot, I'm really glad that for whatever reason this old thread suddenly popped up on my "foryou" page here on GitHub, even if for reasons unknown… 

That's a very useful link regardless.
